src/vs/editor/contrib/inlineCompletions/ghostTextController.ts

```diff
@@ -54,29 +54,41 @@ export class GhostTextController extends Disposable {
 		this.updateModelController();
 	}
 
+	// Don't call this method when not neccessary. It will recreate the activeController.
 	private updateModelController(): void {
 		const suggestOptions = this.editor.getOption(EditorOption.suggest);
 
 		this.activeController.value = undefined;
-		this.activeController.value = this.editor.hasModel() && suggestOptions.showSuggestionPreview
-			? this.instantiationService.createInstance(ActiveGhostTextController, this.editor, this.widget, this.contextKeys)
-			: undefined;
+		// ActiveGhostTextController is only created if one of those settings is set or if the inline completions are triggered explicitly.
+		this.activeController.value =
+			this.editor.hasModel() && (suggestOptions.showSuggestionPreview || suggestOptions.showInlineCompletions || this.triggeredExplicitly)
+				? this.instantiationService.createInstance(
+					ActiveGhostTextController,
+					this.editor,
+					this.widget,
+					this.contextKeys
+				)
+				: undefined;
 	}
 
 	public shouldShowHoverAt(hoverRange: Range): boolean {
 		return this.activeController.value?.shouldShowHoverAt(hoverRange) || false;
 	}
 
 	public trigger(): void {
-		this.activeController.value?.trigger();
+		this.triggeredExplicitly = true;
```

Comment:
Shouldn't this be set to `false` somewhere when the session ends?

Reply:
This is just for the lazy initialization of the active controller (which will subscribe to the suggest widget and what not). That controller should not be disposed when the session ends. The session itself has checks that it is only triggered automatically when the settings allow it.